### PR TITLE
Migrate cw721-base to cosmwasm 1.0

### PR DESCRIPTION
**Pull request by measure-fi**
https://github.com/palomachain/paloma/issues/177

## Background

Migrate cw721 to cosmwasm 1.0

## Testing completed

- [x] Existing tests pass.

### CODE REVIEW

File: contracts/cw721-base/src/state.rs [hunk 1]
```diff
@@ -123,7 +143,7 @@ where
     T: Serialize + DeserializeOwned + Clone,
 {
     // pk goes to second tuple element
-    pub owner: MultiIndex<'a, (Addr, Vec<u8>), TokenInfo<T>>,
+    pub owner: MultiIndex<'a, (Addr, Vec<u8>), TokenInfo<T>, &'a str>,
```
Comment (measure-fi) [inline contracts/cw721-base/src/state.rs:126]: So this and line 160 below, I'm significantly less confident this is the correct transformation. I need to come back to this and walk through the two query functions that depend on it.

Adding an issue to do this.

File: contracts/cw721-base/src/query.rs [hunk 2]
```diff
@@ -65,20 +65,25 @@ where
     ) -> StdResult<OperatorsResponse> {
         let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
         let start_addr = maybe_addr(deps.api, start_after)?;
-        let start = start_addr.map(|addr| Bound::exclusive(addr.as_ref()));
+        let start = start_addr.as_ref().map(Bound::exclusive);
 
         let owner_addr = deps.api.addr_validate(&owner)?;
-        let res: StdResult<Vec<_>> = self
+        let operators: Vec<_> = self
```
Comment (Vizualni) [inline contracts/cw721-base/src/query.rs]: Out of curiosity, what does `Vec<_>` means (the underscore part)?

**Review by Vizualni — APPROVED** [review]

File: contracts/cw721-base/src/state.rs (hunk 1 above)
Comment (measure-fi) [reply contracts/cw721-base/src/state.rs:126]: https://github.com/palomachain/paloma/issues/214

File: contracts/cw721-base/src/query.rs (hunk 2 above)
Comment (measure-fi) [reply contracts/cw721-base/src/query.rs]: `collect()` in Rust is a common way to end iterator chains, it collects the elements into anything that implements `FromIterator`. There are a couple examples in the [docs for collect](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect).

That genericity means that it often needs to have a type hint, do we want to collect into a `Vec` or a `HashMap` or a `Result` or ..? But the type hint does not need to be complete, `_` asks to just infer what this should be a `Vec` of.

**Pull request closed without merging**